### PR TITLE
Update tests for OSX

### PR DESCRIPTION
**Pull request by eam**

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: src/test/func-test/test_misc_func.cc [hunk 1]
```diff
@@ -143,9 +143,14 @@ class TestMiscFunc : public TestMiscFuncSuit {
   CPPUNIT_TEST(test_misc_func_31);
   CPPUNIT_TEST(test_misc_func_32);
   CPPUNIT_TEST(test_misc_func_33);
+#ifdef __linux__
+// windowed mode not supported on non-linux
   CPPUNIT_TEST(test_misc_func_34);
+#endif
   CPPUNIT_TEST(test_misc_func_35);
+#ifdef __linux__
   CPPUNIT_TEST(test_misc_func_36);
```
Comment (areese) [inline src/test/func-test/test_misc_func.cc]: can you add a comment for what this test is that's not supported on non-linux?
Is it also windowed mode?

File: src/test/pool_test/test_ccmp.cc [hunk 2]
```diff
@@ -26,10 +26,13 @@ class YccmpTest: public CppUnit::TestFixture {
 
   CPPUNIT_TEST_SUITE(YccmpTest);
    
+#ifdef __linux__
+// uses /proc, does not exist on OSX
```
Comment (areese) [inline src/test/pool_test/test_ccmp.cc]: Shouldn't this be #ifndef _DARWIN_ or something.
There should be some sort of /proc on the later bsd's.  (I don't have a later bsd handy(

**Comment by areese** [comment]
looks good to me.
It'd be nice to treat it as 3 OSes instead of just linux vs OS-X.

**Comment by eam** [comment]
There are a few tests that absolutely are linux-only; I figured I'd err on the side of disabling with the assumption that these have mostly been developed for linux. Don't have a FreeBSD box handy to test on, unfortunately.

I'll switch it around to darwin-only excludes.

**Comment by timrc-git** [comment]
Several thoughts:
a) this pretty much clinches that I should add a feature flag to disable windowed-mode
b) my apologies for not disabling the robust-lock tests on that feature flag
c) if possible, I'd like to pull the makefile differences into Makefile.base
    Or just unconditionally set DYLD_LIBRARY_PATH, or use RPATH/RUNPATH

I'll take a crack at this later (need to get some other stuff done first).

Thanks!
.timrc

**Comment by eam** [comment]
Oh, of course. I should've wrapped those in `HAVE_ROBUST_PTHREADS`.

**Comment by eam** [comment]
I moved the LD_LIB_PATH bits into `Makefile.base`, commented the reasons for excluding tests and switched to using `HAVE_ROBUST_PTHREADS`. I think this is ready to go.

**Comment by timrc-git** [comment]
Thanks Evan!

After I merge this, I plan to add a flag for windowed mode. 
Hopefully, I can twist your arm to retest then.

**Comment by eam** [comment]
Absolutely.

**Pull request merged**